src/nvme/log.c

```diff
@@ -81,7 +83,7 @@ __nvme_msg(nvme_root_t r, int lvl,
 		message = NULL;
 	va_end(ap);
 
-	if (lvl <= r->log_level)
+	if (!r || lvl <= r->log_level)
```

Comment:
Shouldn't we test that the message is at the right level at the beginning of the function and return right away if we're not going to print it? This would avoid performing a whole lot of operations for nothing.

In other words:
```
/* Check if message is at the right level. If not, get out immediately */
if (r && lvl > r->log_level)  
    return;
```

Reply:
Good point. I'll update it accordingly :)